frontend/ui/elements/common_settings_menu_table.lua

```diff
@@ -541,14 +542,23 @@ common_settings.document = {
 
 You can decide between two locations where these will be saved:
-- alongside the book file itself (the long time default): these sdr folders will be visible when you browse your library directories with another file browser or from your computer, which may clutter your vision of your library. But this allows you to move them along when you reorganize your library, and also survives any renaming of parent directories. Also, if you perform directory synchronization or backups, your settings will be part of them.
-- all inside koreader/docsettings/: these sdr folders will only be visible and used by KOReader, and won't clutter your vision of your library directories with another file browser or from your computer. But any reorganisation of your library (directories or filename moves and renamings) may result in KOReader not finding your previous settings for these books. These settings won't be part of any synchronization or backups of your library.]])
+- alongside the book file itself (the long time default): sdr folders will be visible when you browse your library directories with another file browser or from your computer, which may clutter your vision of your library. But this allows you to move them along when you reorganize your library, and also survives any renaming of parent directories. Also, if you perform directory synchronization or backups, your settings will be part of them.
+- all in koreader/docsettings/: sdr folders will only be visible and used by KOReader, and won't clutter your vision of your library directories with another file browser or from your computer. But any reorganisation of your library (directories or filename moves and renamings) may result in KOReader not finding your previous settings for these books. These settings won't be part of any synchronization or backups of your library.
+- all inside koreader/hashdocsettings/ as hashes: sdr folders are identified not by filepath/filename but by partial MD5 hash, allowing you to rename, move, and copy documents outside of KOReader without sdr folder clutter while keeping them linked to their metadata. However, any file modifications such as writing highlights into PDFs or downloading from Calibre may change the hash, and thus lose their linked metadata. Calculating file hashes may also slow down file browser navigation. This option may suit users with multiple copies of documents across different devices and directories.]])
+
+
+local hash_filemod_warn = _([[Warning: /koreader/hashdocsettings requires calculating partial file hashes of documents which may slow down file browser navigation. Any file modifications (such as embedding annotations into PDF files or downloading from Calibre) may change its partial hash, resulting in the metadata being unlinked from the file and thereby losing its highlights, bookmarks, and progress. Embedding PDF annotations is currently set to "%s" and can be disabled at (Cog -> Document -> Save Document (write highlights into PDF)).]])
```

Comment:
I don't really like the false implications of `/koreader`. Couldn't we just populate this with the actual path instead of something the user has to resolve for themselves regardless?

Reply:
This was copied from the previously-existing lines above. But this has `/` and the others don't, which might be interpreted as absolute. I could remove the `/`?

Reply:
The warning can be put in the form of an icon in the popup.

Don't use `->`, use `→`. Much prettier. ;-)

```suggestion
local hash_filemod_warn = T(_([[$1 requires calculating partial file hashes of documents which may slow down file browser navigation. Any file modifications (such as embedding annotations into PDF files or downloading from Calibre) may change the partial hash, thereby losing track of any highlights, bookmarks, and progress data. Embedding PDF annotations is currently set to "%s" and can be disabled at (⚙ → Document → Save Document (write highlights into PDF)).]]), Data:getStorage().."/hashstuff")
```

Reply:
I didn't even know the cog unicode existed, neat. Updating.

Reply:
Just realized `Data:getStorage` doesn't exist, is there a good way to get the absolute path somewhere?

Reply:
It exists in the same sense that `hashstuff` does. DataStorage:getDataDir() and friends. In fact depending on how often you do this it might be worth adding a getHashDir() to it.

Reply:
Got it, and apparently I already did add that function so we're good. Will update the others.

Reply:
Also I just noticed I accidentally wrote $1 rather than %1. Unlike the stuff at the end that wasn't on purpose.

Reply:
Yes, just koreader (or ./koreader) should be fine. It's mainly the / I was commenting on.